manage

```diff
@@ -708,22 +719,30 @@ Or on Discord: \e[94m\e[4mhttps://discord.gg/RXSTvaUvuu\e[0m" | tee -a "$logfile
   #exit the manage script with the same exit-code of the app's script
   exit $exitcode
   
-elif [ "$1" == 'update' ] && [[ "$2" =~ ^filelist:.* ]];then #list of files separated by :
-  # use the second argument to pass our list of files to update
-  #get functions from updater script
-  source "${DIRECTORY}/updater" source
-
-  updatable_apps='' updatable_files="$(echo "$2" | sed 's/^filelist://g' | tr ':' '\n')" no_status=true update_now_cli
+elif [ "$1" == 'refresh' ];then #copy the online version of an app to the main pi-apps folder. Used when an app does not need to be reinstalled when updating it/
+  status -n "Refreshing \e[1m${app}\e[0m\e[96m... "
+  #Set terminal title
+  echo -ne "\e]0;Refreshing ${app}\a"
+  
+  #move old program to trash - this allows app-developers to recover their work if update was accidental
+  gio trash "${DIRECTORY}/apps/${app}" 2>/dev/null
+  #failsafe
+  rm -rf "${DIRECTORY}/apps/${app}"
+  
+  #copy new version from update/ to apps/
+  cp -rf "${DIRECTORY}/update/pi-apps/apps/${app}" "${DIRECTORY}/apps/${app}"
   
-elif [ "$1" == 'update' ] || [ "$1" == 'refresh' ];then
+  status_green Done
+  
+elif [ "$1" == 'update' ];then #user-facing argument to update an app. Pi-Apps scripts should directly use updater's update_app function.
   #for this operation, a program name must be specified.
   app="$2"
   if [ -z "$app" ];then
     error "For this operation, you must specify which app to operate on."
   fi
   
-  # make sure update_app function is available
-  typeset -f update_app &>/dev/null || source "${DIRECTORY}/updater" source
+  #get functions from updater script
+  source "${DIRECTORY}/updater" source
```

Comment:
this seems to be in conflict with your changes above that made sure the `updater` was already sourced. what is the point of making sure it was sourced if you don't use it at all for `refresh` and source it anyway for `update` actions?

Reply:
Good catch. I had briefly tried having manage-daemon directly use updater' functions for both the `update` and `refresh` actions as a time-saving measure. I later decided against it as it created duplicate codepaths. I will remove the residual `$sourced_updater` code shortly.

Reply:
Oh wait. It is necessary because updating files uses `update_now_cli`.